### PR TITLE
[IndexTable]: Fix IndexTable sticky header overlap

### PR DESCRIPTION
**Pull request by alllx**
### WHY are these changes introduced?
On the smaller screens IndexTable with sticky first column has issue with not overlapping other headers when scrolled horizontally (see screenshot below 👇🏼)

Screenshot 2023-05-11 at 6 17 44 PM

Hey @samrose3 could you please check that this fix do not break what you were fixed as looks like this bug introduced as a regression after this fix https://github.com/Shopify/polaris/pull/9119 

<details>
<summary>Before</summary>

https://github.com/Shopify/polaris/assets/701295/44a3fdc2-e613-4dd7-9ec4-5f7045470785

</details>

<details>
<summary>After</summary>

https://github.com/Shopify/polaris/assets/701295/366f5b91-3d2d-4c0f-82ff-f91fd7e77aeb

</details>

### WHAT is this pull request doing?
Adding `z-index` to the `.TableHeading-first` and `.TableHeading-second` to set overlapping of other headers when scrolling horizontally.

### How to 🎩
Open http://localhost:6006/?path=/story/all-components-indextable--with-sortable-headings to see the `IndexTable` component with sticky columns.
Spin instance shop1 https://admin.web.shipify-labels-beta-51js.alexander-yamkov.us.spin.dev/admin/spin/tophat?activate=true
Go to `/store/shop1/shipping_labels` or `/store/shop1/orders` page, reduce width to see horizontal scroll in the table.

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md) ~ N/A
- [ ] Updated the component's `README.md` with documentation changes ~ N/A
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide ~ N/A

### CODE REVIEW

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 1]
```diff
@@ -437,6 +439,8 @@ $loading-panel-height: 53px;
       @media #{$p-breakpoints-sm-up} {
         position: sticky;
         left: 0;
+        // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
```
Comment (sam-b-rose) [inline polaris-react/src/components/IndexTable/IndexTable.scss]:
```suggestion
        // stylelint-disable-next-line -- second column (first heading item) overlaps other headings on horizontal scroll
```

File: polaris-react/src/components/IndexTable/IndexTable.scss [hunk 2]
```diff
@@ -270,6 +270,8 @@ $loading-panel-height: 53px;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   padding-right: var(--pc-index-table-checkbox-offset-right);
   width: var(--p-space-5);
+  // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
```
Comment (sam-b-rose) [inline polaris-react/src/components/IndexTable/IndexTable.scss]:
```suggestion
  // stylelint-disable-next-line -- first column (checkbox) overlaps other headings on horizontal scroll
```

**Review by sam-b-rose — APPROVED** [review]
🎩 looks good! Fixes the regrestion and retains the fix from #9119 💯

I left a few suggestions adding context to the Stylelint disable comments.

Before merging, we should also add a changeset to describe the fix and apply the patch update (run `yarn changeset` to add a new changeset) 🦋

**Review by chloerice — APPROVED** [review]
🎩 Tophat looks great, thanks for jumping on this @alllx! 🙌🏽  Just need to add a [change log entry](https://github.com/Shopify/polaris/blob/main/.changeset/README.md) and this is good to ship 🚢

**Pull request merged**